### ISSUE TITLE
Compilation problem

### ISSUE THREAD

**Issue by morsmordreb** [NVIDIA/go-nvml#102]
How can I know the relationship between the current tag of our project and the cuda version? There are many errors when compiling. For example, I am currently using CUDA-12.0, corresponding to the project 12.0-0, there will be such errors:

expected ‘nvmlDevice_t’ {aka ‘struct <anonymous>’} but argument is of type ‘struct nvmlDevice_st *’
nvmlReturn_t DECLDIR nvmlVgpuTypeGetMaxInstances(nvmlDevice_t device, nvmlVgpuTypeId_t vgpuTypeId, unsigned int *vgpuInstanceCount);

Here's what makfie wrote
CGO_CFLAGS = -i/usr/local/cuda - 12.0 / include
CGO_LDFLAGS="-L /usr/local/cuda-12.0/lib64-lnvidia-ml"

**Comment by elezar** [comment]
@morsmordreb you should be able to use newer bindings against older CUDA versions since we detect the versioned APIs automatically. (There may be some edge cases that are not covered, but we can address those if reported).

It is even possible to use older bindings against newer CUDA versions -- with the caveat that any functions added since then will not be available.

With this in mind, I don't think it's required to provide your own `nvml.h` to build applications that use this package.

**Comment by rahulait** [comment]
This issue has been open for over 90 days without recent updates, and the context may now be outdated.

To keep the issue tracker clean and focused on current and actionable topics, I am going to close this issue. If you are still blocked by this or have more queries, please feel free to re-open it or create a new issue so that someone from team can take a look.
